### PR TITLE
[Mobile Payments] Play sound and activate Taptic feedback when a payment is completed successfully

### LINKED ISSUE #4193: [Mobile Payments] Sounds!
Add a sound when the payment has been completed

### PR DESCRIPTION
**Pull request by ctarda**
Closes #4193 

https://user-images.githubusercontent.com/2722505/119652354-faf51380-bdf3-11eb-9340-1832b0305d07.mov

## Changes
* Play our favourite cha-ching sound when a payment is completed successfully 
* Also trigger haptic feedback for those boring souls, like myself, that always have their devices in silent mode.

## How to test
Testing requires:

1. Setting up a store for Card Present Payment Testing (also P91TBi-4BH-p2 for more specific instructions)
2. WCPay 2.4.0

* Checkout the branch, build and run on device.
* Connect to a reader.
* Hold the device in your hand. Turn the volume up! 🔉 
* Collect a payment for an eligible order. When the process is completed successfully, the device should provide haptic feedback and a sound should be played 💸 

cc @astralbodies 

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by koke** [comment]
It's been a long time since I dealt with sound on iOS, but I remembered there was a specific API meant for this kind of short alert sound: [Audio Services](https://developer.apple.com/documentation/audiotoolbox/audio_services).

I made a [quick prototype](https://github.com/koke/ChaChing/) to test this (SwiftUI is really cool for this), and the AVFoundation based player would stop any other sounds that iOS is playing, while Audio Services will play the alert on top of any currently playing sounds:


https://user-images.githubusercontent.com/8739/119677151-c35f8900-be3e-11eb-9473-327445e33a49.MOV

**Comment by ctarda** [comment]
Thanks @koke  I just pushed an update to play the sound using the AudioToolbox APIs. 🔊

File: WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureCelebration.swift [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import AudioToolbox
+import UIKit
+
+/// Plays a sound and provides taptic feedback when a payment capture has been completed successfully
```
Comment (allendav) [inline WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureCelebration.swift]: taptic -> haptic

File: WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureCelebration.swift [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+/// https://www.youtube.com/watch?v=ewRjZoRtu0Y
```
Comment (allendav) [inline WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureCelebration.swift]: Why this link? We aren't using this sound, are we?

Comment (ctarda) [reply WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureCelebration.swift]: Nah. It was a joke. Maybe it was not a funny joke. I will remove it.

File: WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureCelebration.swift [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        }
+
+        let url = URL(fileURLWithPath: path)
+        AudioServicesCreateSystemSoundID(url as CFURL, &soundID)
+        AudioServicesPlaySystemSound(soundID)
+    }
+
+    func shakeDevice() {
+        let generator = UINotificationFeedbackGenerator()
+        generator.prepare()
```
Comment (allendav) [inline WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureCelebration.swift]: I can't target my phone today to try this out, but does this prepare make a noticeable difference?

**Review by allendav — APPROVED** [review]
Tests fine on iPad. Unable to test on iPhone today.  Assuming that works, :shipit:

File: WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureCelebration.swift [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+private extension PaymentCaptureCelebration {
+    func playSound() {
+        guard let path = Bundle.main.path(forResource: "o.caf", ofType: nil) else {
+            return
+        }
+
+        let url = URL(fileURLWithPath: path)
+        AudioServicesCreateSystemSoundID(url as CFURL, &soundID)
```
Comment (koke) [inline WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureCelebration.swift:21]: I think you're only meant to call this once for each sound in your application, or dispose of the system sound when you're done with it with `AudioServicesDisposeSystemSoundID(_:)`

Comment (ctarda) [reply WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureCelebration.swift:21]: Nice catch thanks!

**Comment by ctarda** [comment]
@koke I just pushed an update to release the sound. Would you mind taking another look, please?

**Review by koke — APPROVED** [review]
I'd be slightly more comfortable with initializing once per session, but I can't anticipate any actual issues with normal usage, and this seems to work fine. Let's just keep that in mind if we start playing sound effects for more things.

**Pull request merged**